From 2585851b3e426eaacfd0700296eddde10cab88cc Mon Sep 17 00:00:00 2001
From: Claire Girka <claire@sitedethib.com>
Date: Fri, 15 Jul 2022 22:09:57 +0200
Subject: [PATCH 3/9] [pending review] mmdevapi: copy ContainerID from audio
 driver if available.

Some games with support for the haptic feedback and speaker features of the
Sony DualSense controller select the controller's audio output by filtering on
the ContainerId IMMDevice property to find one that matches the controller's
HID's.
---
 dlls/mmdevapi/devenum.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/mmdevapi/devenum.c b/dlls/mmdevapi/devenum.c
index 8a427d347d9..8a39886f918 100644
--- a/dlls/mmdevapi/devenum.c
+++ b/dlls/mmdevapi/devenum.c
@@ -421,6 +421,8 @@ static MMDevice *MMDevice_Create(const WCHAR *name, GUID *id, EDataFlow flow, DW
             MMDevice_SetPropValue(id, flow, (const PROPERTYKEY*)&DEVPKEY_DeviceInterface_FriendlyName, &pv);
             MMDevice_SetPropValue(id, flow, (const PROPERTYKEY*)&DEVPKEY_Device_DeviceDesc, &pv);
 
+            set_driver_prop_value(id, flow, (const PROPERTYKEY*)&DEVPKEY_Device_ContainerId);
+
             pv.pwszVal = guidstr;
             MMDevice_SetPropValue(id, flow, &deviceinterface_key, &pv);
 
-- 
2.42.2

